cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -108,11 +107,8 @@ public GoalOptimizer(KafkaCruiseControlConfig config,
         _defaultModelCompletenessRequirements.minMonitoredPartitionsPercentage(),
         _defaultModelCompletenessRequirements.includeAllTopics());
     _goalByPriorityForPrecomputing = new ArrayList<>();
-    // The number of proposal precomputing thread should not exceed the number of unique goal priority combinations.
-    _numPrecomputingThreads = Math.min(config.getInt(KafkaCruiseControlConfig.NUM_PROPOSAL_PRECOMPUTE_THREADS_CONFIG),
-                                       AnalyzerUtils.factorial(_goalsByPriority.size()));
-    // Generate different goal priorities for each of proposal precomputing thread.
-    populateGoalByPriorityForPrecomputing();
+    _numPrecomputingThreads = config.getInt(KafkaCruiseControlConfig.NUM_PROPOSAL_PRECOMPUTE_THREADS_CONFIG);
+    _goalByPriorityForPrecomputing.add(new ArrayList<>(_goalsByPriority));
```

Comment:
Maybe we can remove this field since now it is always of size 1 and contains `_goalsByPriority`.